simple-dns/src/dns/rdata/txt.rs

```diff
@@ -117,6 +115,38 @@ impl<'a> TryFrom<HashMap<String, Option<String>>> for TXT<'a> {
     }
 }
 
+impl<'a> TryFrom<String> for TXT<'a> {
+    type Error = crate::SimpleDnsError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let mut txt = TXT::new();
+
+        let mut start_index = 0;
+        let full_length = value.len();
+
+        while start_index < full_length {
+            let end_index = (start_index + MAX_CHARACTER_STRING_LENGTH).min(full_length);
+
+            let slice = &value[start_index..end_index];
+            txt.add_char_string(slice.to_string().try_into()?);
+
+            start_index = end_index;
+        }
+
+        Ok(txt)
+    }
+}
+
+impl<'a> From<TXT<'a>> for String {
+    fn from(val: TXT<'a>) -> Self {
+        val.strings
+            .into_iter()
+            .map(|s| s.into())
+            .collect::<Vec<String>>()
+            .join("")
+    }
+}
```

Comment:
Here would be the proper place to check for a valid utf8 string, something like this
```suggestion
impl<'a> TryFrom<TXT<'a>> for String {
    type Error = std::string::FromUtf8Error;
    fn try_from(val: TXT<'a>) -> Result<Self, Self::Error> {
        let bytes = val
            .strings
            .into_iter()
            .fold(Vec::with_capacity(val.len()), |mut acc, val| {
                acc.extend(val.data.as_ref());
                acc
            });

        String::from_utf8(bytes)
    }
}

```